src/guiFormSpecMenu.cpp

```diff
@@ -1887,6 +1982,12 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 		GUITable *table = m_tables[i].second;
 		mydata.table_dyndata[tablename] = table->getDynamicData();
 	}
+	
+	//preserve editboxes
+	for (u32 i=0; i<m_fields.size(); i++) {
+		if(getElementFromId(m_fields[i].fid)->getTypeName() != std::string("editBox")) continue;
```

Comment:
Spaces after `if` and around operators; also do not put the body on the same line.